tests/parser/test-synced-patterns.php

```diff
@@ -226,6 +226,273 @@ public function test_synced_pattern_with_override() {
+
+		// Core block attributes can change, so we use assertArraySubset to avoid
+		// brittle assertions.
+		$this->assertArraySubset( $expected_blocks, $blocks['blocks'], false, wp_json_encode( $blocks['blocks'] ) );
+		$this->assertEquals( 1, count( $blocks['blocks'] ), 'Too many blocks in result set' );
+		$this->assertEquals( 1, count( $blocks['blocks'][0]['innerBlocks'] ), 'Too many inner blocks in synced pattern' );
+	}
+
+	/* Synced pattern with sourced attributen nested content */
+
+	public function test_synced_pattern_with_sourced_attribute_in_nested_content() {
+		$this->register_block_with_attributes( 'test/custom-block', [
+			'content' => [
+				'type'               => 'rich-text',
+				'source'             => 'rich-text',
+				'selector'           => 'p',
+				'__experimentalRole' => 'content',
+			],
+			'bing'    => [
+				'type'      => 'string',
+				'source'    => 'attribute',
+				'selector'  => 'p',
+				'attribute' => 'data-bing',
+			],
+		] );
+
+		$synced_pattern_content = '
+			<!-- wp:group -->
+			<div class="wp-block-group">
+				<!-- wp:group -->
+				<div class="wp-block-group">
+					<!-- wp:test/custom-block -->
+					<p data-bing="bong">My nested synced pattern content</p>
+					<!-- /wp:test/custom-block -->
+				</div>
+				<!-- /wp:group -->
+			</div>
+			<!-- /wp:group -->
+		';
+
+		$synced_pattern = $this->factory()->post->create_and_get( [
+			'post_content' => $synced_pattern_content,
+			'post_status'  => 'publish',
+			'post_type'    => 'wp_block',
+		] );
+
+		$html = sprintf( '<!-- wp:block {"ref":%d} /-->', $synced_pattern->ID );
+
+		$expected_blocks = [
+			[
+				'name'        => 'core/block',
+				'attributes'  => [
+					'ref' => $synced_pattern->ID,
+				],
+				'innerBlocks' => [
+					[
+						'name'        => 'core/group',
+						'attributes'  => [
+							'tagName' => 'div',
+						],
+						'innerBlocks' => [
+							[
+								'name'        => 'core/group',
+								'attributes'  => [
+									'tagName' => 'div',
+								],
+								'innerBlocks' => [
+									[
+										'name'       => 'test/custom-block',
+										'attributes' => [
+											'content' => 'My nested synced pattern content',
+											'bing'    => 'bong',
+										],
+									],
+								],
+							],
+						],
+					],
+				],
+			],
+		];
+
+		$content_parser = new ContentParser( $this->get_block_registry() );
+		$blocks         = $content_parser->parse( $html );
+
+		$this->assertArrayHasKey( 'blocks', $blocks, sprintf( 'Unexpected parser output: %s', wp_json_encode( $blocks ) ) );
+		$this->assertEquals( $expected_blocks, $blocks['blocks'], sprintf( 'Blocks not equal: %s', wp_json_encode( $blocks['blocks'] ) ) );
+	}
+
+	/* Synced pattern with override in nested content */
+
+	public function test_synced_pattern_with_override_in_nested_content() {
+		$synced_pattern_content = '
+			<!-- wp:group -->
+			<div class=k"wp-block-group">
```

Comment:
Minor note, there's a 3 `k"`s scattered in here that appear to be an accident, unless I'm missing something.

Reply:
Fixed in https://github.com/Automattic/vip-block-data-api/pull/87/commits/5dbe30812dd750bfd8414e184b97c06dff687776

#vimfail